### PR TITLE
OSEM reconstruction example

### PR DESCRIPTION
**Pull request by robbietuk**

### CODE REVIEW

**Comment by robbietuk** [comment]
This PR is pretty much ready. `ExampleReconstruction.sh` will run when the script is given Prompts, addative sinogram, and multfactors.

**Comment by robbietuk** [comment]
Is there anything else I can add to this PR?

File: ExampleReconstruction/ExampleReconstruction.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+## AUTHOR: ROBERT TWYMAN
+## Copyright (C) 2021 University College London
+## Licensed under the Apache License, Version 2.0
+
+
+# ExampleReconstruction.sh script perform OSEM as an example reconstruction for GATE data.
+#  The required inputs are as follow:
+#		Coincidence Data 
+#		Additive Sinogram 
+#		MultiplicativeF actors"
```
Comment (KrisThielemans) [inline ExampleReconstruction/ExampleReconstruction.sh:11]:
```suggestion
#		MultiplicativeFactors
```

File: ExampleReconstruction/OSEM.par [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    recompute sensitivity := 1
+
+    xy output image size (in pixels) := -1
+
+  end PoissonLogLikelihoodWithLinearModelForMeanAndProjData Parameters:=
+
+  ; Number of subsets should usually be a divisor of num_views/8
+  output filename prefix := ${SGCPATH}/ExampleReconstruction/OSEM
+  number of subsets:= 8
+  number of subiterations:= 40
```
Comment (KrisThielemans) [inline ExampleReconstruction/OSEM.par:36]: provide these as parameters I guess (wouldn't work for the mMR for instance).  Can still hardwire them in the script (but most people will be more comfortable in changing that)

File: ExampleReconstruction/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Created:  Mon 28 Apr 2021
+
+This directory contains the files for PET image reconstruction using STIR.
+
+
+Files
+=======
+
+* `OSEM.par`: Example STIR OSEM reconstuction parameter file. The file contains enviromental variable.
```
Comment (KrisThielemans) [inline ExampleReconstruction/README.md:23]:
```suggestion
* `OSEM.par`: Example STIR OSEM reconstruction parameter file. The file uses environment variables.
```

File: ExampleReconstruction/README.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+
+Scripts
+=======
+
+* `ExampleOSEM.sh`: An example script that exports the required enviromental variables for image reconstruction using `OSEM.par`.
```
Comment (KrisThielemans) [inline ExampleReconstruction/README.md:29]:
```suggestion
* `ExampleOSEM.sh`: An example script that exports the required enviroment variables for image reconstruction using `OSEM.par`.
```

File: README.md [hunk 5]
```diff
@@ -48,6 +48,7 @@ Directories
 * `ExamplesOfScannersMacros/`: Contains examples of macros for creating a scanner in GATE.
 * `ExamplesOfPhantomMacros/`: Contains examples of macros and STIR parameter files for creating/importing source and attenuation phantoms.
 * `VoxelisedSimulation/`: Main GATE simulation directory.
+* `ExampleReconstuction/`: Contains example scripts for iterative reconstruction using STIR, e.g. OSEM.
```
Comment (KrisThielemans) [inline README.md:51]:
```suggestion
* `ExampleReconstruction/`: Contains example scripts for iterative reconstruction using STIR, e.g. OSEM.
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

**Comment by robbietuk** [comment]
Wow I am sorry, my spelling is appauling today.

File: ExampleReconstruction/OSEM.par (hunk 2 above)
Comment (robbietuk) [reply ExampleReconstruction/OSEM.par]: I moved it out to the script with the option to set them

**Comment by robbietuk** [comment]
The comments/documentation of the script and parameter file are qutie verbose. The idea being users might not be that familiar with STIR.

**Comment by robbietuk** [comment]
Possible overlap with #75 when we consider `forward_projector_proj_matrix_ray_tracing.par`. It should be noted in the reconstruction that we use the same projector configuration as the data correction projector.

**Pull request merged**